File: javaScript/Algorithms/binarysearch.js
function binarySearch(list, item) {
    var min = 0
    var max = list.length - 1
    var mid
    while (min <= max) {
        mid = Math.floor((min + max) / 2)
        if (list[mid] === item) {
            return `${item} found at index ${mid}`
        }
        else {
            if (item < list[mid]) {
                max = mid - 1
            }
            else {
                min = mid + 1
            }
        }
    }
    return `${item} is not in list`
}

console.log(binarySearch([1,2,3,4,5,6,7,8,9], 3))

File: C++/binarysearch.cpp
/* Implemention of binary search algorithm in C++ */

#include <iostream>

using namespace std;

int binarySearch(int item_list[], int item) {
	int first = 0;
	int last = sizeof(item_list)-1;
	bool found = false;
	while (first <= last && !found) {
		int mid = (first + last)/2;
		if (item_list[mid] == item) {
			found = true;
		} else {
			if (item < item_list[mid]) { last = mid - 1; }
			else { first = mid + 1; }
		}
	}
	return found;
}

int main()
{
    // Examples of use
    int arrayOne[] = {1,2,3,5,8};
    cout << binarySearch(arrayOne, 6) << endl;
    int arrayTwo[] = {1,2,3,5,8};
    cout << binarySearch(arrayTwo, 5) << endl;
}
